src/osd/PrimaryLogPG.cc

```diff
@@ -7110,6 +7185,45 @@ void PrimaryLogPG::start_copy(CopyCallback *cb, ObjectContextRef obc,
   _copy_some(obc, cop);
 }
 
+void PrimaryLogPG::start_repair_copy(OpContext *ctx, uint32_t what,
+				     const vector<pg_shard_t>& bad_shards)
+{
+  const auto& dest = ctx->obs->oi.soid;
+  auto found = copy_ops.find(dest);
+  if (found != copy_ops.end()) {
+    cancel_copy(found->second, false);
+  }
+  // we expect the client side to prepend an assert_version() call before
+  // this op, so just pass 0 as the version.
+  auto cb = new CopyFromCallback(ctx);
+  ctx->copy_cb = cb;
+  uint32_t to_copy = 0;
+  if (what & librados::repair_copy_t::DATA)
```

Comment:
I'd consider those other uses bugs - the fadvise flags in fact have CEPH_OSD_OP_FLAG...* constants defined, and that's important since that header is shared with the kernel.

The other current user is the 2nd librados listing interface, which imo is another misuse of the librados namespace, since it's used in the wire protocol for pgnls. 

We shouldn't be able to break the wire protocol accidentally by changing a librados type.